{"ast":null,"code":"export function getGeneralApiProblem(response){switch(response.problem){case\"CONNECTION_ERROR\":return{kind:\"cannot-connect\",temporary:true};case\"NETWORK_ERROR\":return{kind:\"cannot-connect\",temporary:true};case\"TIMEOUT_ERROR\":return{kind:\"timeout\",temporary:true};case\"SERVER_ERROR\":return{kind:\"server\"};case\"UNKNOWN_ERROR\":return{kind:\"unknown\",temporary:true};case\"CLIENT_ERROR\":switch(response.status){case 401:return{kind:\"unauthorized\"};case 403:return{kind:\"forbidden\"};case 404:return{kind:\"not-found\"};default:return{kind:\"rejected\"};}case\"CANCEL_ERROR\":return null;}return null;}","map":{"version":3,"names":["getGeneralApiProblem","response","problem","kind","temporary","status"],"sources":["/home/ehsan/dockerize-expo-rn-web-app/PizzaApp/app/services/api/apiProblem.ts"],"sourcesContent":["import { ApiResponse } from \"apisauce\"\n\nexport type GeneralApiProblem =\n  /**\n   * Times up.\n   */\n  | { kind: \"timeout\"; temporary: true }\n  /**\n   * Cannot connect to the server for some reason.\n   */\n  | { kind: \"cannot-connect\"; temporary: true }\n  /**\n   * The server experienced a problem. Any 5xx error.\n   */\n  | { kind: \"server\" }\n  /**\n   * We're not allowed because we haven't identified ourself. This is 401.\n   */\n  | { kind: \"unauthorized\" }\n  /**\n   * We don't have access to perform that request. This is 403.\n   */\n  | { kind: \"forbidden\" }\n  /**\n   * Unable to find that resource.  This is a 404.\n   */\n  | { kind: \"not-found\" }\n  /**\n   * All other 4xx series errors.\n   */\n  | { kind: \"rejected\" }\n  /**\n   * Something truly unexpected happened. Most likely can try again. This is a catch all.\n   */\n  | { kind: \"unknown\"; temporary: true }\n  /**\n   * The data we received is not in the expected format.\n   */\n  | { kind: \"bad-data\" }\n\n/**\n * Attempts to get a common cause of problems from an api response.\n *\n * @param response The api response.\n */\nexport function getGeneralApiProblem(response: ApiResponse<any>): GeneralApiProblem | null {\n  switch (response.problem) {\n    case \"CONNECTION_ERROR\":\n      return { kind: \"cannot-connect\", temporary: true }\n    case \"NETWORK_ERROR\":\n      return { kind: \"cannot-connect\", temporary: true }\n    case \"TIMEOUT_ERROR\":\n      return { kind: \"timeout\", temporary: true }\n    case \"SERVER_ERROR\":\n      return { kind: \"server\" }\n    case \"UNKNOWN_ERROR\":\n      return { kind: \"unknown\", temporary: true }\n    case \"CLIENT_ERROR\":\n      switch (response.status) {\n        case 401:\n          return { kind: \"unauthorized\" }\n        case 403:\n          return { kind: \"forbidden\" }\n        case 404:\n          return { kind: \"not-found\" }\n        default:\n          return { kind: \"rejected\" }\n      }\n    case \"CANCEL_ERROR\":\n      return null\n  }\n\n  return null\n}\n"],"mappings":"AA6CA,MAAO,SAAS,CAAAA,oBAAoBA,CAACC,QAA0B,CAA4B,CACzF,OAAQA,QAAQ,CAACC,OAAO,EACtB,IAAK,kBAAkB,CACrB,MAAO,CAAEC,IAAI,CAAE,gBAAgB,CAAEC,SAAS,CAAE,IAAK,CAAC,CACpD,IAAK,eAAe,CAClB,MAAO,CAAED,IAAI,CAAE,gBAAgB,CAAEC,SAAS,CAAE,IAAK,CAAC,CACpD,IAAK,eAAe,CAClB,MAAO,CAAED,IAAI,CAAE,SAAS,CAAEC,SAAS,CAAE,IAAK,CAAC,CAC7C,IAAK,cAAc,CACjB,MAAO,CAAED,IAAI,CAAE,QAAS,CAAC,CAC3B,IAAK,eAAe,CAClB,MAAO,CAAEA,IAAI,CAAE,SAAS,CAAEC,SAAS,CAAE,IAAK,CAAC,CAC7C,IAAK,cAAc,CACjB,OAAQH,QAAQ,CAACI,MAAM,EACrB,IAAK,IAAG,CACN,MAAO,CAAEF,IAAI,CAAE,cAAe,CAAC,CACjC,IAAK,IAAG,CACN,MAAO,CAAEA,IAAI,CAAE,WAAY,CAAC,CAC9B,IAAK,IAAG,CACN,MAAO,CAAEA,IAAI,CAAE,WAAY,CAAC,CAC9B,QACE,MAAO,CAAEA,IAAI,CAAE,UAAW,CAAC,CAC/B,CACF,IAAK,cAAc,CACjB,MAAO,KAAI,CACf,CAEA,MAAO,KAAI,CACb"},"metadata":{},"sourceType":"module","externalDependencies":[]}